js/src/wrappers/anthropic.ts

```diff
@@ -218,8 +252,7 @@ function streamNextProxy(stream: AsyncIterator<any>, sspan: StartedSpan) {
     }
 
     if (result.done) {
-      totals.tokens =
-        (totals["prompt_tokens"] || 0) + (totals["completion_tokens"] || 0);
+      totals = finalizeMetrics(totals);
```

Comment:
I know I wrote this but it's worth double checking that we always get a "done" result. In some other async tracing code I moved `span.end` up one level because you might not make if there's an error or something.

If so, we might want to do the summing each loop (would require keeping a bit more state) or push it up a level. I can help you look into this if so. Or .. you can say "this is as good as it was before I got here" and I can look it into it when you're done.

Reply:
I think we should investigate this separately and probably write some tests that prove whether or not it would fail. I'll let you take this!